beacon-chain/forkchoice/protoarray/store.go

```diff
@@ -230,6 +230,29 @@ func (s *Store) NodesIndices() map[[32]byte]uint64 {
 	return s.nodesIndices
 }
 
+// ChainHeads returns all possible chain heads (leaves of fork choice tree).
+// Heads roots and heads slots are returned.
+func (s *Store) ChainHeads() ([][32]byte, []types.Slot) {
+	s.nodesLock.RLock()
+	nodes := s.Nodes()
+	defer s.nodesLock.RUnlock()
```

Comment:
oops this should not be `defer`

```suggestion
	s.nodesLock.RUnlock()
```